front/src/common/RollingStockSelector/RollingStockCardDetail.tsx

```diff
@@ -40,36 +55,42 @@ export default function RollingStockCardDetail(props) {
       };
       // AC & HEATING curves (optional)
       curvesData.modes[mode].curves.forEach((curve) => {
-        const serieId = `${name} ${curve.cond.comfort}`;
-        transformedCurves[serieId] = { ...curve.curve, mode: name, comfort: curve.cond.comfort };
+        if (curve.cond?.comfort) {
+          const optionalCurveName = `${name} ${curve.cond.comfort}`;
+          transformedCurves[optionalCurveName] = {
+            ...curve.curve,
+            mode: name,
+            comfort: curve.cond.comfort as string,
+          };
+        }
       });
     });
     return transformedCurves;
   };
 
-  const getRollingStock = async () => {
-    try {
-      const rollingStockData = await get(`${ROLLING_STOCK_URL + id}/`);
-      setData(rollingStockData);
-      setCurvesComfortList(listCurvesComfort(rollingStockData.effort_curves));
-    } catch (e) {
+  const { data: rollingStock, error } = osrdEditoastApi.useGetRollingStockByIdQuery(
+    { id },
+    {
+      skip: !id,
+    }
+  );
```

Comment:
This is weird, the rollingStock should be called in the parent component. We will change this here, but can you add a comment ?
```suggestion
  // TODO: move the fetch of the rollingStock in the parent component
  const { data: rollingStock, error } = osrdEditoastApi.useGetRollingStockByIdQuery(
    { id },
    {
      skip: !id,
    }
  );
```

Reply:
done